pymc/printing.py

```diff
@@ -130,148 +102,194 @@ def str_for_model(model: Model, formatting: str = "plain", include_params: bool
+        )
+    if formatting == "latex":
+        name = _latex_clean_command(name, command="text")
+        dist_name = _latex_clean_command(dist_name, command="operatorname")
+
+    # Arguments passed to the distribution or expression
+    if _has_owner(var) and isinstance(var.owner.op, RandomVariable):
+        # var is the RV or dist() from a Distribution.
+        dist_args = var.owner.inputs[3:]  # First 3 inputs are always rng, size, dtype
+    elif _has_owner(var) and isinstance(var.owner.op, SymbolicRandomVariable):
+        # var is a symbolic RV from a Distribution.
+        dist_args = [
+            x
+            for x in var.owner.inputs
+            if not isinstance(x, (RandomStateSharedVariable, RandomGeneratorSharedVariable))
+        ]
     else:
-        return _str_for_expression(var, formatting)
-
+        # Assume that var is a Deterministic or a Potential.
+        dist_args = _walk_expression_args(var)
+    args_str = _str_for_args_list(dist_args, formatting=formatting)
+    if _is_potential_or_deterministic(var):
+        args_str = f"f({args_str})"
 
-def _str_for_input_rv(var: Variable, formatting: str) -> str:
-    _str = (
-        var.name
-        if var.name is not None
-        else str_for_dist(var, formatting=formatting, include_params=True)
-    )
-    if "latex" in formatting:
-        return _latex_text_format(_latex_escape(_str.strip("$")))
-    else:
-        return _str
+    # These three strings are now formatted according to `formatting`. If latex, they
+    # just ultimately need to be wrapped in $.
+    return name, dist_name, args_str
 
 
-def _str_for_constant(var: Union[Constant, SharedVariable], formatting: str) -> str:
-    if isinstance(var, Constant):
-        var_data = var.data
-        var_type = "constant"
-    else:
-        var_data = var.get_value()
-        var_type = "shared"
-
-    if len(var_data.shape) == 0:
-        return f"{var_data:.3g}"
-    elif len(var_data.shape) == 1 and var_data.shape[0] == 1:
-        return f"{var_data[0]:.3g}"
-    elif "latex" in formatting:
-        return rf"\text{{<{var_type}>}}"
+def _str_for_input_var(var: Variable, formatting: str) -> str:
+    """Make a human-readable string representation for a variable that
+    serves as input to another variable."""
+    # Check for constants first, because they won't have var.owner
+    if isinstance(var, (Constant, SharedVariable)):
+        # Give the constant value if it's small enough, else basic type info.
+        # Previously _str_for_constant()
+        if isinstance(var, Constant):
+            var_data = var.data
+            var_type = "constant"
+        else:
+            var_data = var.get_value()
+            var_type = "shared"
+        if var_data.size == 1:
+            return f"{var_data.flatten()[0]:.3g}"
+        else:
+            return f"<{var_type} {var_data.shape}>"
+    elif _has_owner(var):
+        if isinstance(var.owner.op, DimShuffle):
```

Comment:
Still have this question. Is the DimShuffle Op case still one that we need to handle? If so, let's add a test for it.

Reply:
I see, cool. What would be an example of a model where the original RV gets resized or dimshuffled? (@ricardoV94 )